module/reports/helper.js

```diff
@@ -0,0 +1,479 @@
+/**
+ * name : helper.js
+ * author : Rakesh
+ * created-date : 01-Nov-2020
+ * Description : Entity report functionalities
+ */
+
+// Dependencies
+
+const UserProjectsHelper = require(MODULES_BASE_PATH + "/userProjects/helper");
+const mongoose = require('mongoose');
+const moment = require('moment');
+
+
+
+/**
+    * Report Helper
+    * @class
+*/
+
+module.exports = class ReportsHelper {
+
+    /**
+    * Entity Report.
+    * @method
+    * @name entity 
+    * @param {String} entityId - mapped entity id.
+    * @param {String} userId - Logged in user id.
+    * @param {String} reportType - report type monthly or quterly.
+    * @param {String} programId - program id
+    * @returns {Object} Entity report.
+   */
+    static entity(entityId = "", userId, reportType, programId = "",getPdf=false) {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                let query = {};
+                if (entityId) {
+                    query["entityInformation._id"] = mongoose.Types.ObjectId(entityId);
```

Comment:
@aman-tunerlabs - Would suggest having an entityId and entityExternalId outside in user projects and also in index.

Cc - @rakeshSgr